projects/plugins/jetpack/extensions/blocks/donations/custom-amount/edit.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * WordPress dependencies
+ */
+import { InnerBlocks } from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
+import { useContext } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import DonationsContext from '../common/context';
+
+const Edit = () => {
+	const { showCustomAmount, currency } = useContext( DonationsContext );
+
+	return (
+		showCustomAmount && (
+			<InnerBlocks
+				templateLock={ 'all' }
+				template={ [
+					[
+						'core/paragraph',
+						{
+							content: __( 'Or enter a custom amount', 'jetpack' ),
+						},
+					],
+					[
+						'core/group',
```

Comment:
I've used a row block here but I have doubts if it would be better to use the buttons block instead. Semantically it makes more sense to use a row, but we have some styling options that don't apply to our custom amount field. The buttons block better fits our styling needs but is semantically incorrect.

Reply:
Is there a reason for this at all? 🤔 

Like, couldn't we just simply add the Amount block in the root of the template, without wrapping it into Group or Buttons?

Reply:
If Gutenberg supported per-block locking, we could also potentially not need this Custom Amount block at all, instead, we could just move the Amount block directly to the View inner blocks.
(I'll talk more about this elsewhere 😄).

Reply:
The reason for adding the custom amount block inside a Group or Buttons is so that we can have layout options. Since now the donation amount buttons can be re-arranged I think we also need the custom amount to support this. A user might want to arrange the buttons and custom amount on the right side or centered, for example;

<img width="681" alt="Captura de pantalla 2022-02-03 a las 21 13 44" src="https://user-images.githubusercontent.com/1989914/152421605-1720663d-e023-438e-b4b0-3ca74d7a6a88.png">

Or like:

<img width="684" alt="Captura de pantalla 2022-02-03 a las 21 14 18" src="https://user-images.githubusercontent.com/1989914/152421663-67292038-156b-40a8-a915-fc8d5705b26a.png">

Reply:
> If Gutenberg supported per-block locking, we could also potentially not need this Custom Amount block at all, instead, we could just move the Amount block directly to the View inner blocks.

That would be nice, I didn't find a way in which I could achieve conditional adding/removing of a block inside an InnerBlocks component during runtime (without using wp.data), so I had to create the Custom Amount block.